### ISSUE TITLE
Handling of `nthreads`

### ISSUE THREAD

**Issue by dkobak** [KlugerLab/FIt-SNE#40]
I am a bit confused by how `nthreads` parameter is treated.

First, what's the reason for passing it as an argument to `main()` instead of writing it into a file?

Second, if no parameter is passed, then `nthreads=0` by default. This is clearly unusable so in this case you do 

```
    if (nthreads == 0) {
        nthreads = std::thread::hardware_concurrency();
    }
```

but this is done separately in each function that uses `nthreads` as opposed to in `main()`. Why?

Third, the above snippet is _not_ included into `evaluateError` and `evaluateErrorFft`, is it just a mistake? Should we maybe move that snippet into `main()` from everywhere?

Also, fourth, I noticed that there are several parallel loops that do not use `PARALLEL_FOR`; is it on purpose?

**Comment by linqiaozhi** [comment]
1) It was implemented this way while ago in #6, and I never got around to changing it. You're right though, there is no reason why this parameter should passed differently than the others.

2 and 3) Fixed in #41. This was a particularly important catch, as the computeError() were not multithreading at all, as you reported in #37. On my machine with 12 cores (2 threads each), this takes the compute error from about 15 seconds to 1 second, as it parallelizes very nicely. 

4) After  #41, the only loops that do not use `PARALLEL_FOR` are the ones where we have a progress bar that should only output on a single thread (it checks if `t==0`). I don't know how to do that in a way that works for both OpenMP and C++11 threads, so I just hardcode it as C++11 threads for now. Should be changed in the future though.

Thanks @dkobak!

**Comment by dkobak** [comment]
OK great! I think (1) is not important and it can stay like that. I close this.
